src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/RcaController.java

```diff
@@ -277,13 +276,14 @@ private String getElectedMasterHostAddress() {
     return "";
   }
 
-  private void handleNodeRoleChange(
-      final NodeDetails currentNode, final String electedMasterHostAddress) {
+  private void handleNodeRoleChange(final NodeDetails currentNode) {
     final NodeRole currentNodeRole = NodeRole.valueOf(currentNode.getRole());
-    if (currentNode.getHostAddress().equalsIgnoreCase(electedMasterHostAddress)) {
-      currentRole = NodeRole.ELECTED_MASTER;
+    Boolean isMasterNode = currentNode.getIsMasterNode();
+    if (isMasterNode != null) {
+      currentRole = isMasterNode ? NodeRole.ELECTED_MASTER : currentNodeRole;
     } else {
-      currentRole = currentNodeRole;
+      final String electedMasterHostAddress = getElectedMasterHostAddress();
+      currentRole = currentNode.getHostAddress().equalsIgnoreCase(electedMasterHostAddress) ? NodeRole.ELECTED_MASTER : currentNodeRole;
```

Comment:
Because this is going to make ES API call, perhaps we should have a log for now and then we can add a metric for this.